core/modules/layout/layout.api.php

```diff
@@ -196,6 +196,206 @@ function hook_layout_renderer_info() {
 }
 
 /**
+ * Respond to a block being removed from a layout.
+ *
+ * This hook is invoked from Layout::removeBlock() after a block has been 
+ * removed from a layout.
+ * In the context of the Layout Builder User Interface, this hook is 
+ * executed when a user clicks the "Remove block" link in the dropbutton
+ * which displays the block edit options.
+ * In a custom module, if custom settings have been saved as a result of
+ * a hook_block_save() operation, hook_layout_block_remove() allows
+ * these settings to be deleted if the block is removed before the layout
+ * is saved. 
+ *
+ * @param Layout $layout
+ *   The layout object from which the block is being removed.
+ * @param Block $block
+ *   The block object which is being removed.
+ *
+ * @see hook_block_save()
+ * @see hook_block_add()
+ *
+ * @ingroup layout_api_hooks
+ */
+function hook_layout_block_remove(Block $block, Layout $layout) {
+  if ($block->delta == 'my_block') {
+    // Block was removed from layout, so delete custom settings.
+    config_set('my_module.settings', 'custom_settings', array());
+  }
+}
+
+/**
+ * Respond to a block being added to a layout.
+ *
+ * This hook is invoked from Layout::addBlock() after a block has been 
+ * programmatically added to a region in a layout, and also from 
+ * layout_block_configure_form_submit() when a block is added to a layout in the
+ * Layout Builder UI.
```

Comment:
Hm, okay so this was changed so that it doesn't sound so confusing, but now it's inaccurate. This isn't called from `layout_block_configure_form_submit()` at all.